src/target/riscv/riscv-013.c

```diff
@@ -686,6 +715,25 @@ static int dmi_write_exec(struct target *target, uint32_t address,
 	return dmi_op(target, NULL, NULL, DMI_OP_WRITE, address, value, true, ensure_success);
 }
 
+static bool dbgbase_verify(struct target *target)
+{
+	uint32_t dbgbase_set = target->dbgbase;
+	uint32_t nextdm = 0;
+
+	while (1) {
+		target->dbgbase = nextdm;
+		if (target->dbgbase == dbgbase_set)
+			return true;
+		if (dmi_read(target, &nextdm, DM_NEXTDM) != ERROR_OK)
+			break;
+		LOG_DEBUG("dm_nextdm=0x%x", nextdm);
```

Comment:
I've tried to improve the error message - it would be good to display both the old and new address for easier troubleshooting.

```suggestion
		LOG_TARGET_DEBUG(target, "dm @ 0x%x --> nextdm=0x%x", target->dbgbase, nextdm);
```